Qemu builder crash with Packer 1.6.5

#### Overview of the Issue

As of Packer 1.6.5, my qemu build crashes with `panic: interface conversion: interface {} is nil, not int`

This works with 1.6.3 and 1.6.4 for me.

#### Reproduction Steps

Attempt to build with Packer 1.6.5

### Packer version

Packer v1.6.5

### Simplified Packer Buildfile

I'm using a modified version of the chef bento buildfile.  The scripts are just the scripts from bento, and are also irrelevant for this bug.

<details>
  <summary>Packer Buildfile</summary>
  
```
{
  "builders": [
    {
      "boot_key_interval": "10ms",
      "boot_command": [
        "<esc><wait500ms>",
        "<esc><wait500ms>",
        "<enter><wait500ms>",
        "/install/vmlinuz<wait500ms>",
        " console=ttyS0,115200<wait500ms>",
        " auto<wait500ms>",
        " console-setup/ask_detect=false<wait500ms>",
        " console-setup/layoutcode=us<wait500ms>",
        " console-setup/modelcode=pc105<wait500ms>",
        " debconf/frontend=noninteractive<wait500ms>",
        " debian-installer=en_US.UTF-8<wait500ms>",
        " fb=false<wait500ms>",
        " initrd=/install/initrd.gz<wait500ms>",
        " kbd-chooser/method=us<wait500ms>",
        " keyboard-configuration/layout=USA<wait500ms>",
        " keyboard-configuration/variant=USA<wait500ms>",
        " locale=en_US.UTF-8<wait500ms>",
        " netcfg/get_domain=vm<wait500ms>",
        " netcfg/get_hostname=vagrant<wait500ms>",
        " grub-installer/bootdev=/dev/vda<wait500ms>",
        " noapic<wait500ms>",
        " preseed/url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/{{user `preseed_path`}}<wait500ms>",
        " -- <wait500ms>",
        "<enter><wait500ms>"
      ],
      "boot_wait": "10s",
      "cpus": "{{ user `cpus` }}",
      "disk_size": "{{user `disk_size`}}",
      "headless": "{{ user `headless` }}",
      "http_directory": "{{user `http_directory`}}",
      "iso_checksum": "{{user `iso_checksum`}}",
      "iso_url": "{{user `mirror`}}/{{user `mirror_directory`}}/{{user `iso_name`}}",
      "memory": "{{ user `memory` }}",
      "output_directory": "{{ user `build_directory` }}",
      "shutdown_command": "echo 'vagrant' | sudo -S shutdown -P now",
      "ssh_password": "vagrant",
      "ssh_port": 22,
      "ssh_timeout": "10000s",
      "ssh_username": "vagrant",
      "type": "qemu",
      "vm_name": "{{ user `template` }}",
      "machine_type": "q35",
      "net_device": "virtio-net-pci",
      "net_bridge": "br10",
      "ssh_host": "192.168.10.70",
      "qemuargs": [
        [ "-serial", "file:{{ user `template` }}-serial.log" ]
      ]
    }
  ],
  "provisioners": [
    {
      "environment_vars": [
        "HOME_DIR=/home/vagrant",
        "http_proxy={{user `http_proxy`}}",
        "https_proxy={{user `https_proxy`}}",
        "no_proxy={{user `no_proxy`}}"
      ],
      "execute_command": "echo 'vagrant' | {{.Vars}} sudo -S -E sh -eux '{{.Path}}'",
      "expect_disconnect": true,
      "scripts": [
        "{{template_dir}}/scripts/update.sh",
        "{{template_dir}}/scripts/sshd.sh",
        "{{template_dir}}/scripts/networking.sh",
        "{{template_dir}}/scripts/sudoers.sh",
        "{{template_dir}}/scripts/vagrant.sh",
        "{{template_dir}}/scripts/sercons.sh",
        "{{template_dir}}/scripts/cleanup.sh"
      ],
      "type": "shell"
    }
  ],
  "variables": {
    "box_basename": "ubuntu-20.04",
    "build_directory": "builds",
    "build_timestamp": "{{isotime \"20060102150405\"}}",
    "cpus": "2",
    "disk_size": "5000",
    "git_revision": "__unknown_git_revision__",
    "guest_additions_url": "",
    "headless": "true",
    "http_directory": "{{template_dir}}/http",
    "http_proxy": "{{env `http_proxy`}}",
    "https_proxy": "{{env `https_proxy`}}",
    "iso_checksum": "f11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2",
    "iso_name": "ubuntu-20.04.1-legacy-server-amd64.iso",
    "memory": "1024",
    "mirror": "http://cdimage.ubuntu.com",
    "mirror_directory": "ubuntu-legacy-server/releases/20.04.1/release",
    "name": "ubuntu-20.04",
    "no_proxy": "{{env `no_proxy`}}",
    "preseed_path": "preseed.cfg",
    "template": "ubuntu-20.04-amd64",
    "version": "TIMESTAMP"
  }
}
```
</details>

### Operating system and Environment details

Arch Linux:
```
Linux r320 5.4.72-1-lts #1 SMP Sat, 17 Oct 2020 13:30:57 +0000 x86_64 GNU/Linux
```

Qemu:
```
QEMU emulator version 5.1.0
Copyright (c) 2003-2020 Fabrice Bellard and the QEMU Project developers
```

### Log Fragments and crash.log files

<details>
  <summary>packer build output</summary>
  
```
2020/11/01 14:33:13 [INFO] Packer version: 1.6.5 [go1.15.3 linux amd64]
2020/11/01 14:33:13 Checking 'PACKER_CONFIG' for a config file path
2020/11/01 14:33:13 'PACKER_CONFIG' not set; checking the default config file path
2020/11/01 14:33:13 Attempting to open config file: /home/vond/.packerconfig
2020/11/01 14:33:13 [WARN] Config file doesn't exist: /home/vond/.packerconfig
2020/11/01 14:33:13 Setting cache directory: /storage/packer-qemu/ubuntu/packer_cache
2020/11/01 14:33:13 Creating plugin client for path: /usr/local/bin/packer
2020/11/01 14:33:13 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-builder-qemu"}
2020/11/01 14:33:13 Waiting for RPC address for: /usr/local/bin/packer
2020/11/01 14:33:13 packer-builder-qemu plugin: [INFO] Packer version: 1.6.5 [go1.15.3 linux amd64]
2020/11/01 14:33:13 packer-builder-qemu plugin: Checking 'PACKER_CONFIG' for a config file path
2020/11/01 14:33:13 packer-builder-qemu plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/11/01 14:33:13 packer-builder-qemu plugin: Attempting to open config file: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-builder-qemu plugin: [WARN] Config file doesn't exist: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-builder-qemu plugin: Setting cache directory: /storage/packer-qemu/ubuntu/packer_cache
2020/11/01 14:33:13 packer-builder-qemu plugin: args: []string{"packer-builder-qemu"}
2020/11/01 14:33:13 packer-builder-qemu plugin: Plugin address: unix /tmp/packer-plugin206546435
2020/11/01 14:33:13 packer-builder-qemu plugin: Waiting for connection...
2020/11/01 14:33:13 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin206546435
2020/11/01 14:33:13 packer-builder-qemu plugin: Serving a plugin connection...
2020/11/01 14:33:13 Creating plugin client for path: /usr/local/bin/packer
2020/11/01 14:33:13 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-shell"}
2020/11/01 14:33:13 Waiting for RPC address for: /usr/local/bin/packer
2020/11/01 14:33:13 packer-provisioner-shell plugin: [INFO] Packer version: 1.6.5 [go1.15.3 linux amd64]
2020/11/01 14:33:13 packer-provisioner-shell plugin: Checking 'PACKER_CONFIG' for a config file path
2020/11/01 14:33:13 packer-provisioner-shell plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/11/01 14:33:13 packer-provisioner-shell plugin: Attempting to open config file: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-provisioner-shell plugin: Setting cache directory: /storage/packer-qemu/ubuntu/packer_cache
2020/11/01 14:33:13 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2020/11/01 14:33:13 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin355876270
2020/11/01 14:33:13 packer-provisioner-shell plugin: Waiting for connection...
2020/11/01 14:33:13 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin355876270
2020/11/01 14:33:13 packer-provisioner-shell plugin: Serving a plugin connection...
2020/11/01 14:33:13 Preparing build: qemu
2020/11/01 14:33:13 packer-builder-qemu plugin: use detected accelerator: kvm
2020/11/01 14:33:13 Build debug mode: false
2020/11/01 14:33:13 Force build: true
2020/11/01 14:33:13 On error: 
2020/11/01 14:33:13 Waiting on builds to complete...
qemu: output will be in this color.
2020/11/01 14:33:13 Starting build run: qemu
2020/11/01 14:33:13 Running builder: qemu

2020/11/01 14:33:13 [INFO] (telemetry) Starting builder qemu
2020/11/01 14:33:13 packer-builder-qemu plugin: Qemu path: /usr/bin/qemu-system-x86_64, Qemu Image page: /usr/bin/qemu-img
==> qemu: Retrieving ISO
==> qemu: Trying http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso
2020/11/01 14:33:13 packer-builder-qemu plugin: Acquiring lock for: http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso?checksum=sha256%3Af11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2 (/storage/packer-qemu/ubuntu/packer_cache/48e4ec4daa32571605576c5566f486133ecc271f.iso.lock)
==> qemu: Trying http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso?checksum=sha256%3Af11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2
2020/11/01 14:33:19 packer-builder-qemu plugin: Leaving retrieve loop for ISO
==> qemu: http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso?checksum=sha256%3Af11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2 => /storage/packer-qemu/ubuntu/packer_cache/48e4ec4daa32571605576c5566f486133ecc271f.iso
2020/11/01 14:33:19 packer-builder-qemu plugin: No floppy files specified. Floppy disk will not be made.
==> qemu: Deleting previous output directory...
2020/11/01 14:33:19 packer-builder-qemu plugin: No CD files specified. CD disk will not be made.
2020/11/01 14:33:19 packer-builder-qemu plugin: [INFO] Creating disk with Path: builds/ubuntu-20.04-amd64 and Size: 5000M
==> qemu: Creating required virtual machine disks
2020/11/01 14:33:19 packer-builder-qemu plugin: Executing qemu-img: []string{"create", "-f", "qcow2", "builds/ubuntu-20.04-amd64", "5000M"}
2020/11/01 14:33:19 packer-builder-qemu plugin: stdout: Formatting 'builds/ubuntu-20.04-amd64', fmt=qcow2 cluster_size=65536 compression_type=zlib size=5242880000 lazy_refcounts=off refcount_bits=16
2020/11/01 14:33:19 packer-builder-qemu plugin: stderr:
2020/11/01 14:33:19 packer-builder-qemu plugin: Found available port: 8772 on IP: 0.0.0.0
==> qemu: Starting HTTP server on port 8772
    qemu: net_bridge is set; skipping port forwarding setup.
2020/11/01 14:33:19 packer-builder-qemu plugin: Looking for available port between 5900 and 6000 on 127.0.0.1
==> qemu: Looking for available port between 5900 and 6000 on 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Found available port: 5900 on IP: 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Found available VNC port: 5900 on IP: 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Qemu --version output: QEMU emulator version 5.1.0
2020/11/01 14:33:19 packer-builder-qemu plugin: Copyright (c) 2003-2020 Fabrice Bellard and the QEMU Project developers
2020/11/01 14:33:19 packer-builder-qemu plugin: Qemu version: 5.1.0
==> qemu: Starting VM, booting from CD-ROM
    qemu: view the screen of the VM, connect via VNC without a password to
    qemu: vnc://127.0.0.1:0
2020/11/01 14:33:19 packer-builder-qemu plugin: Qemu Builder has no floppy files, not attaching a floppy.
    qemu: The VM will be run headless, without a GUI. If you want to
    qemu: view the screen of the VM, connect via VNC without a password to
    qemu: vnc://127.0.0.1:0
2020/11/01 14:33:19 packer-builder-qemu plugin: failed to unlock port lockfile: close tcp 127.0.0.1:5900: use of closed network connection
==> qemu: Overriding default Qemu arguments with qemuargs template option...
Build 'qemu' errored after 5 seconds 887 milliseconds: unexpected EOF

==> Wait completed after 5 seconds 887 milliseconds

==> Some builds didn't complete successfully and had errors:
--> qemu: unexpected EOF

==> Builds finished but no artifacts were created.
2020/11/01 14:33:19 packer-builder-qemu plugin: panic: interface conversion: interface {} is nil, not int
2020/11/01 14:33:19 packer-builder-qemu plugin: 
2020/11/01 14:33:19 packer-builder-qemu plugin: goroutine 255 [running]:
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).applyUserOverrides(0xc0001ca060, 0xc000a5e8a0, 0xc000668e00, 0x535df40, 0xc0001cc1e0, 0x5, 0xc000a35f40, 0xc000888380, 0xc000888320, 0xc0004173c0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:287 +0xfdf
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).getCommandArgs(0xc0001ca060, 0xc000668e00, 0x535df40, 0xc0001cc1e0, 0x0, 0x0, 0xc0001cc1e8, 0xc0001cc210, 0x4d1addc)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:370 +0x85
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).Run(0xc0001ca060, 0x535d300, 0xc0001d0040, 0x535df40, 0xc0001cc1e0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:47 +0x37b
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc0001cc240, 0x535d300, 0xc0001d0040, 0x535df40, 0xc0001cc1e0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/helper/multistep/basic_runner.go:70 +0x234
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*Builder).Run(0xc000668e00, 0x535d300, 0xc0001d0040, 0x53785e0, 0xc0001cc090, 0x52e9320, 0xc0001ca040, 0xc000980000, 0xc000185418, 0xc00094cbd0, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/builder.go:158 +0x11a2
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc000816b40, 0x1, 0xc000416fd0, 0x0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/packer/rpc/builder.go:117 +0x1c4
2020/11/01 14:33:19 packer-builder-qemu plugin: reflect.Value.call(0xc0001853e0, 0xc00018a2e0, 0x13, 0x4d10ab6, 0x4, 0xc00094cf08, 0x3, 0x3, 0x0, 0x0, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/reflect/value.go:475 +0x8c7
2020/11/01 14:33:19 packer-builder-qemu plugin: reflect.Value.Call(0xc0001853e0, 0xc00018a2e0, 0x13, 0xc000824708, 0x3, 0x3, 0x0, 0x0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/reflect/value.go:336 +0xb9
2020/11/01 14:33:19 packer-builder-qemu plugin: net/rpc.(*service).call(0xc000816b80, 0xc0000ca410, 0xc00050e560, 0xc00050e570, 0xc00014d600, 0xc00000dde0, 0x3e8fda0, 0xc000416fbc, 0x18a, 0x3d92a80, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/net/rpc/server.go:377 +0x189
2020/11/01 14:33:19 packer-builder-qemu plugin: created by net/rpc.(*Server).ServeCodec
panic: interface conversion: interface {} is nil, not int
2020/11/01 14:33:19 packer-builder-qemu plugin: 
2020/11/01 14:33:19 packer-builder-qemu plugin: goroutine 255 [running]:
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).applyUserOverrides(0xc0001ca060, 0xc000a5e8a0, 0xc000668e00, 0x535df40, 0xc0001cc1e0, 0x5, 0xc000a35f40, 0xc000888380, 0xc000888320, 0xc0004173c0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:287 +0xfdf
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).getCommandArgs(0xc0001ca060, 0xc000668e00, 0x535df40, 0xc0001cc1e0, 0x0, 0x0, 0xc0001cc1e8, 0xc0001cc210, 0x4d1addc)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:370 +0x85
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).Run(0xc0001ca060, 0x535d300, 0xc0001d0040, 0x535df40, 0xc0001cc1e0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:47 +0x37b
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc0001cc240, 0x535d300, 0xc0001d0040, 0x535df40, 0xc0001cc1e0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/helper/multistep/basic_runner.go:70 +0x234
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*Builder).Run(0xc000668e00, 0x535d300, 0xc0001d0040, 0x53785e0, 0xc0001cc090, 0x52e9320, 0xc0001ca040, 0xc000980000, 0xc000185418, 0xc00094cbd0, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/builder.go:158 +0x11a2
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc000816b40, 0x1, 0xc000416fd0, 0x0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/packer/rpc/builder.go:117 +0x1c4
2020/11/01 14:33:19 packer-builder-qemu plugin: reflect.Value.call(0xc0001853e0, 0xc00018a2e0, 0x13, 0x4d10ab6, 0x4, 0xc00094cf08, 0x3, 0x3, 0x0, 0x0, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/reflect/value.go:475 +0x8c7
2020/11/01 14:33:19 packer-builder-qemu plugin: reflect.Value.Call(0xc0001853e0, 0xc00018a2e0, 0x13, 0xc000824708, 0x3, 0x3, 0x0, 0x0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/reflect/value.go:336 +0xb9
2020/11/01 14:33:19 packer-builder-qemu plugin: net/rpc.(*service).call(0xc000816b80, 0xc0000ca410, 0xc00050e560, 0xc00050e570, 0xc00014d600, 0xc00000dde0, 0x3e8fda0, 0xc000416fbc, 0x18a, 0x3d92a80, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/net/rpc/server.go:377 +0x189
2020/11/01 14:33:19 packer-builder-qemu plugin: created by net/rpc.(*Server).ServeCodec
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/net/rpc/server.go:474 +0x445
2020/11/01 14:33:19 [INFO] (telemetry) ending qemu
2020/11/01 14:33:19 /usr/local/bin/packer: plugin process exited
2020/11/01 14:33:19 ui error: Build 'qemu' errored after 5 seconds 887 milliseconds: unexpected EOF
2020/11/01 14:33:19 ui: 
==> Wait completed after 5 seconds 887 milliseconds
2020/11/01 14:33:19 machine readable: error-count []string{"1"}
2020/11/01 14:33:19 ui error: 
==> Some builds didn't complete successfully and had errors:
2020/11/01 14:33:19 machine readable: qemu,error []string{"unexpected EOF"}
2020/11/01 14:33:19 ui error: --> qemu: unexpected EOF
2020/11/01 14:33:19 ui: 
==> Builds finished but no artifacts were created.
2020/11/01 14:33:19 [INFO] (telemetry) Finalizing.
2020/11/01 14:33:20 waiting for all plugin processes to complete...
2020/11/01 14:33:20 /usr/local/bin/packer: plugin process exited

2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/net/rpc/server.go:474 +0x445
2020/11/01 14:33:19 [INFO] (telemetry) ending qemu
2020/11/01 14:33:19 /usr/local/bin/packer: plugin process exited
==> Wait completed after 5 seconds 887 milliseconds
2020/11/01 14:33:19 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2020/11/01 14:33:19 machine readable: qemu,error []string{"unexpected EOF"}
==> Builds finished but no artifacts were created.
2020/11/01 14:33:19 [INFO] (telemetry) Finalizing.
2020/11/01 14:33:20 waiting for all plugin processes to complete...
2020/11/01 14:33:20 /usr/local/bin/packer: plugin process exited


!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
</details>

<details>
  <summary>crash.log</summary>
  
```
2020/11/01 14:33:13 [INFO] Packer version: 1.6.5 [go1.15.3 linux amd64]
2020/11/01 14:33:13 Checking 'PACKER_CONFIG' for a config file path
2020/11/01 14:33:13 'PACKER_CONFIG' not set; checking the default config file path
2020/11/01 14:33:13 Attempting to open config file: /home/vond/.packerconfig
2020/11/01 14:33:13 [WARN] Config file doesn't exist: /home/vond/.packerconfig
2020/11/01 14:33:13 Setting cache directory: /storage/packer-qemu/ubuntu/packer_cache
2020/11/01 14:33:13 Creating plugin client for path: /usr/local/bin/packer
2020/11/01 14:33:13 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-builder-qemu"}
2020/11/01 14:33:13 Waiting for RPC address for: /usr/local/bin/packer
2020/11/01 14:33:13 packer-builder-qemu plugin: [INFO] Packer version: 1.6.5 [go1.15.3 linux amd64]
2020/11/01 14:33:13 packer-builder-qemu plugin: Checking 'PACKER_CONFIG' for a config file path
2020/11/01 14:33:13 packer-builder-qemu plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/11/01 14:33:13 packer-builder-qemu plugin: Attempting to open config file: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-builder-qemu plugin: [WARN] Config file doesn't exist: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-builder-qemu plugin: Setting cache directory: /storage/packer-qemu/ubuntu/packer_cache
2020/11/01 14:33:13 packer-builder-qemu plugin: args: []string{"packer-builder-qemu"}
2020/11/01 14:33:13 packer-builder-qemu plugin: Plugin address: unix /tmp/packer-plugin206546435
2020/11/01 14:33:13 packer-builder-qemu plugin: Waiting for connection...
2020/11/01 14:33:13 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin206546435
2020/11/01 14:33:13 packer-builder-qemu plugin: Serving a plugin connection...
2020/11/01 14:33:13 Creating plugin client for path: /usr/local/bin/packer
2020/11/01 14:33:13 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-shell"}
2020/11/01 14:33:13 Waiting for RPC address for: /usr/local/bin/packer
2020/11/01 14:33:13 packer-provisioner-shell plugin: [INFO] Packer version: 1.6.5 [go1.15.3 linux amd64]
2020/11/01 14:33:13 packer-provisioner-shell plugin: Checking 'PACKER_CONFIG' for a config file path
2020/11/01 14:33:13 packer-provisioner-shell plugin: 'PACKER_CONFIG' not set; checking the default config file path
2020/11/01 14:33:13 packer-provisioner-shell plugin: Attempting to open config file: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /home/vond/.packerconfig
2020/11/01 14:33:13 packer-provisioner-shell plugin: Setting cache directory: /storage/packer-qemu/ubuntu/packer_cache
2020/11/01 14:33:13 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2020/11/01 14:33:13 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin355876270
2020/11/01 14:33:13 packer-provisioner-shell plugin: Waiting for connection...
2020/11/01 14:33:13 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin355876270
2020/11/01 14:33:13 packer-provisioner-shell plugin: Serving a plugin connection...
2020/11/01 14:33:13 Preparing build: qemu
2020/11/01 14:33:13 packer-builder-qemu plugin: use detected accelerator: kvm
2020/11/01 14:33:13 ui: qemu: output will be in this color.
2020/11/01 14:33:13 ui: 
2020/11/01 14:33:13 Build debug mode: false
2020/11/01 14:33:13 Force build: true
2020/11/01 14:33:13 On error: 
2020/11/01 14:33:13 Waiting on builds to complete...
2020/11/01 14:33:13 Starting build run: qemu
2020/11/01 14:33:13 Running builder: qemu
2020/11/01 14:33:13 [INFO] (telemetry) Starting builder qemu
2020/11/01 14:33:13 packer-builder-qemu plugin: Qemu path: /usr/bin/qemu-system-x86_64, Qemu Image page: /usr/bin/qemu-img
2020/11/01 14:33:13 ui: ==> qemu: Retrieving ISO
2020/11/01 14:33:13 ui: ==> qemu: Trying http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso
2020/11/01 14:33:13 packer-builder-qemu plugin: Acquiring lock for: http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso?checksum=sha256%3Af11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2 (/storage/packer-qemu/ubuntu/packer_cache/48e4ec4daa32571605576c5566f486133ecc271f.iso.lock)
2020/11/01 14:33:13 ui: ==> qemu: Trying http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso?checksum=sha256%3Af11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2
2020/11/01 14:33:19 ui: ==> qemu: http://cdimage.ubuntu.com/ubuntu-legacy-server/releases/20.04.1/release/ubuntu-20.04.1-legacy-server-amd64.iso?checksum=sha256%3Af11bda2f2caed8f420802b59f382c25160b114ccc665dbac9c5046e7fceaced2 => /storage/packer-qemu/ubuntu/packer_cache/48e4ec4daa32571605576c5566f486133ecc271f.iso
2020/11/01 14:33:19 packer-builder-qemu plugin: Leaving retrieve loop for ISO
2020/11/01 14:33:19 ui: ==> qemu: Deleting previous output directory...
2020/11/01 14:33:19 packer-builder-qemu plugin: No floppy files specified. Floppy disk will not be made.
2020/11/01 14:33:19 packer-builder-qemu plugin: No CD files specified. CD disk will not be made.
2020/11/01 14:33:19 ui: ==> qemu: Creating required virtual machine disks
2020/11/01 14:33:19 packer-builder-qemu plugin: [INFO] Creating disk with Path: builds/ubuntu-20.04-amd64 and Size: 5000M
2020/11/01 14:33:19 packer-builder-qemu plugin: Executing qemu-img: []string{"create", "-f", "qcow2", "builds/ubuntu-20.04-amd64", "5000M"}
2020/11/01 14:33:19 packer-builder-qemu plugin: stdout: Formatting 'builds/ubuntu-20.04-amd64', fmt=qcow2 cluster_size=65536 compression_type=zlib size=5242880000 lazy_refcounts=off refcount_bits=16
2020/11/01 14:33:19 packer-builder-qemu plugin: stderr:
2020/11/01 14:33:19 packer-builder-qemu plugin: Found available port: 8772 on IP: 0.0.0.0
2020/11/01 14:33:19 ui: ==> qemu: Starting HTTP server on port 8772
2020/11/01 14:33:19 ui:     qemu: net_bridge is set; skipping port forwarding setup.
2020/11/01 14:33:19 ui: ==> qemu: Looking for available port between 5900 and 6000 on 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Looking for available port between 5900 and 6000 on 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Found available port: 5900 on IP: 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Found available VNC port: 5900 on IP: 127.0.0.1
2020/11/01 14:33:19 packer-builder-qemu plugin: Qemu --version output: QEMU emulator version 5.1.0
2020/11/01 14:33:19 packer-builder-qemu plugin: Copyright (c) 2003-2020 Fabrice Bellard and the QEMU Project developers
2020/11/01 14:33:19 packer-builder-qemu plugin: Qemu version: 5.1.0
2020/11/01 14:33:19 ui: ==> qemu: Starting VM, booting from CD-ROM
2020/11/01 14:33:19 ui:     qemu: The VM will be run headless, without a GUI. If you want to
    qemu: view the screen of the VM, connect via VNC without a password to
    qemu: vnc://127.0.0.1:0
2020/11/01 14:33:19 packer-builder-qemu plugin: Qemu Builder has no floppy files, not attaching a floppy.
2020/11/01 14:33:19 ui: ==> qemu: Overriding default Qemu arguments with qemuargs template option...
2020/11/01 14:33:19 packer-builder-qemu plugin: failed to unlock port lockfile: close tcp 127.0.0.1:5900: use of closed network connection
2020/11/01 14:33:19 packer-builder-qemu plugin: panic: interface conversion: interface {} is nil, not int
2020/11/01 14:33:19 packer-builder-qemu plugin: 
2020/11/01 14:33:19 packer-builder-qemu plugin: goroutine 255 [running]:
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).applyUserOverrides(0xc0001ca060, 0xc000a5e8a0, 0xc000668e00, 0x535df40, 0xc0001cc1e0, 0x5, 0xc000a35f40, 0xc000888380, 0xc000888320, 0xc0004173c0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:287 +0xfdf
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).getCommandArgs(0xc0001ca060, 0xc000668e00, 0x535df40, 0xc0001cc1e0, 0x0, 0x0, 0xc0001cc1e8, 0xc0001cc210, 0x4d1addc)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:370 +0x85
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*stepRun).Run(0xc0001ca060, 0x535d300, 0xc0001d0040, 0x535df40, 0xc0001cc1e0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/step_run.go:47 +0x37b
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc0001cc240, 0x535d300, 0xc0001d0040, 0x535df40, 0xc0001cc1e0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/helper/multistep/basic_runner.go:70 +0x234
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/builder/qemu.(*Builder).Run(0xc000668e00, 0x535d300, 0xc0001d0040, 0x53785e0, 0xc0001cc090, 0x52e9320, 0xc0001ca040, 0xc000980000, 0xc000185418, 0xc00094cbd0, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/builder/qemu/builder.go:158 +0x11a2
2020/11/01 14:33:19 packer-builder-qemu plugin: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc000816b40, 0x1, 0xc000416fd0, 0x0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/home/circleci/project/packer/packer/rpc/builder.go:117 +0x1c4
2020/11/01 14:33:19 packer-builder-qemu plugin: reflect.Value.call(0xc0001853e0, 0xc00018a2e0, 0x13, 0x4d10ab6, 0x4, 0xc00094cf08, 0x3, 0x3, 0x0, 0x0, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/reflect/value.go:475 +0x8c7
2020/11/01 14:33:19 packer-builder-qemu plugin: reflect.Value.Call(0xc0001853e0, 0xc00018a2e0, 0x13, 0xc000824708, 0x3, 0x3, 0x0, 0x0, 0x0)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/reflect/value.go:336 +0xb9
2020/11/01 14:33:19 packer-builder-qemu plugin: net/rpc.(*service).call(0xc000816b80, 0xc0000ca410, 0xc00050e560, 0xc00050e570, 0xc00014d600, 0xc00000dde0, 0x3e8fda0, 0xc000416fbc, 0x18a, 0x3d92a80, ...)
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/net/rpc/server.go:377 +0x189
2020/11/01 14:33:19 packer-builder-qemu plugin: created by net/rpc.(*Server).ServeCodec
2020/11/01 14:33:19 packer-builder-qemu plugin: 	/usr/local/go/src/net/rpc/server.go:474 +0x445
2020/11/01 14:33:19 [INFO] (telemetry) ending qemu
2020/11/01 14:33:19 /usr/local/bin/packer: plugin process exited
2020/11/01 14:33:19 ui error: Build 'qemu' errored after 5 seconds 887 milliseconds: unexpected EOF
2020/11/01 14:33:19 ui: 
==> Wait completed after 5 seconds 887 milliseconds
2020/11/01 14:33:19 machine readable: error-count []string{"1"}
2020/11/01 14:33:19 ui error: 
==> Some builds didn't complete successfully and had errors:
2020/11/01 14:33:19 machine readable: qemu,error []string{"unexpected EOF"}
2020/11/01 14:33:19 ui error: --> qemu: unexpected EOF
2020/11/01 14:33:19 ui: 
==> Builds finished but no artifacts were created.
2020/11/01 14:33:19 [INFO] (telemetry) Finalizing.
2020/11/01 14:33:20 waiting for all plugin processes to complete...
2020/11/01 14:33:20 /usr/local/bin/packer: plugin process exited
```
</details>

--- a/builder/qemu/step_create_disk.go
+++ b/builder/qemu/step_create_disk.go
@@ -32,12 +32,11 @@ func (s *stepCreateDisk) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionContinue
 	}
 
-	var diskFullPaths, diskSizes []string
-
 	ui.Say("Creating required virtual machine disks")
 	// The 'main' or 'default' disk
-	diskFullPaths = append(diskFullPaths, filepath.Join(s.OutputDir, name))
-	diskSizes = append(diskSizes, s.DiskSize)
+	diskFullPaths := []string{filepath.Join(s.OutputDir, name)}
+	diskSizes := []string{s.DiskSize}
+
 	// Additional disks
 	if len(s.AdditionalDiskSize) > 0 {
 		for i, diskSize := range s.AdditionalDiskSize {

--- a/builder/qemu/step_run.go
+++ b/builder/qemu/step_run.go
@@ -216,8 +216,10 @@ func (s *stepRun) getDeviceAndDriveArgs(config *Config, state multistep.StateBag
 			drivesToAttach = append(drivesToAttach, imgPath)
 		}
 
-		diskFullPaths := state.Get("qemu_disk_paths").([]string)
-		drivesToAttach = append(drivesToAttach, diskFullPaths...)
+		if v, ok := state.GetOk("qemu_disk_paths"); ok {
+			diskFullPaths := v.([]string)
+			drivesToAttach = append(drivesToAttach, diskFullPaths...)
+		}
 
 		for i, drivePath := range drivesToAttach {
 			driveArgumentString := fmt.Sprintf("file=%s,if=%s,cache=%s,discard=%s,format=%s", drivePath, config.DiskInterface, config.DiskCache, config.DiskDiscard, config.Format)
@@ -284,7 +286,9 @@ func (s *stepRun) applyUserOverrides(defaultArgs map[string]interface{}, config 
 
 		commHostPort := 0
 		if config.CommConfig.Comm.Type != "none" {
-			commHostPort = state.Get("commHostPort").(int)
+			if v, ok := state.GetOk("commHostPort"); ok {
+				commHostPort = v.(int)
+			}
 		}
 		httpIp := state.Get("http_ip").(string)
 		httpPort := state.Get("http_port").(int)

